packages/core/src/codecs/octetstream-codec.ts

```diff
@@ -66,7 +66,7 @@ export default class OctetstreamCodec implements ContentCodec {
         }
 
         let dataLength = bytes.length;
-        let dataType: string = schema.type;
+        let dataType: string = schema ? schema.type : "undefined";
```

Comment:
I think assignments like these could be replaced like so:

```suggestion
        let dataType: string = schema ?? "undefined";
```